src/main/java/org/dataone/hashstore/hashfs/HashFileStore.java

```diff
@@ -220,4 +219,284 @@ protected HashAddress put(InputStream object, String pid, String additionalAlgor
         return hashAddress;
     }
 
+    public void validateChecksumParameters(String checksum, String checksumAlgorithm) {
+        if (checksum != null && !checksum.trim().isEmpty()) {
+            if (checksumAlgorithm == null) {
+                // ChecksumAlgorithm cannot be null if checksum is supplied
+                logHashFileStore.error(
+                        "HashFileStore.validateChecksumParameters - checksumAlgorithm cannot be null if checksum is supplied");
+                throw new NullPointerException(
+                        "checksumAlgorithm cannot be null if checksum is supplied, checksum and checksumAlgorithm must both be null if validation is not desired.");
+            } else if (checksumAlgorithm.trim().isEmpty()) {
+                // ChecksumAlgorithm cannot be empty when checksum supplied
+                logHashFileStore.error(
+                        "HashFileStore.validateChecksumParameters - checksumAlgorithm cannot be empty if checksum is supplied");
+                throw new IllegalArgumentException(
+                        "checksumAlgorithm cannot be empty when checksum supplied, checksum and checksumAlgorithm must both be null if unused.");
+            } else if (!this.isValidAlgorithm(checksumAlgorithm)) {
+                // ChecksumAlgorithm not supported
+                logHashFileStore.error(
+                        "HashFileStore.validateChecksumParameters - checksumAlgorithm not supported, checksumAlgorithm: "
+                                + checksumAlgorithm);
+                throw new IllegalArgumentException(
+                        "Checksum algorithm not supported - cannot be used to validate object. checksumAlgorithm: "
+                                + checksumAlgorithm + ". Supported algorithms: "
+                                + Arrays.toString(SUPPORTED_HASH_ALGORITHMS));
+            }
+        } else if (checksumAlgorithm != null && !checksumAlgorithm.trim().isEmpty()) {
+            // checksum cannot be null or empty if checksumAlgorithm is supplied
+            if (checksum == null) {
```

Comment:
what about the case where `checksum.trim().isEmpty()`?

Reply:
I have split the check for checksum and checksumAlgorithm into two if statements (more readable now) and added this missing check. In the if statement previously, `checksum.trim().isEmpty()` would always be "true" if it got to the point referenced here in this comment.